File: finalproject(Fontend)/lib/model/payment.dart
import 'package:finalproject/model/membertrip.dart';

class Payment {
  int? paymentId;
  String? paymentStatus;
  double? price;
  String? paymentDetail;
  String? paymentSlip;
  DateTime? datetimePayment;
  MemberTrip? membertrip;

  Payment({
    this.paymentId,
    this.paymentStatus,
    this.price,
    this.paymentDetail,
    this.paymentSlip,
    this.datetimePayment,
    this.membertrip,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
    paymentId: json['paymentId'],
    paymentStatus: json['paymentStatus'] ?? '',
    price: json['price']?.toDouble() ?? 0.0,
    paymentDetail: json['paymentDetail'] ?? '',
    paymentSlip: json['paymentSlip'] ?? '',
    datetimePayment: json['datetimePayment'] != null ? DateTime.parse(json['datetimePayment']) : null,
    membertrip: json['membertrip'] != null ? MemberTrip.fromJson(json['membertrip']) : null,
  );

  Map<String, dynamic> toJson() => {
    'paymentId': paymentId,
    'paymentStatus': paymentStatus,
    'price': price,
    'paymentDetail': paymentDetail,
    'paymentSlip': paymentSlip,
    'datetimePayment': datetimePayment?.toIso8601String(),
    'membertrip': membertrip?.toJson(),
  };
}

File: FinalProject(Backend)/src/main/java/com/finalproject/model/Payment.java
package com.finalproject.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Table(name = "payment")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "paymentId")
    private Integer paymentId;

    @Column(name = "paymentStatus", nullable = false)
    private String paymentStatus;

    @Column(name = "price", nullable = false)
    private Double price;

    @Column(name = "paymentDetail", nullable = false)
    private String paymentDetail;

    @Column(name = "paymentSlip")
    private String paymentSlip;

    @Column(name = "datetimePayment")
    private Date datetimePayment;

    // ใน Payment.java
    @ManyToOne
    @JoinColumn(name = "memberTripId", nullable = false)
    @JsonBackReference
    private MemberTrip membertrip;

}
